posthog/api/team.py

```diff
@@ -300,11 +301,21 @@ def validate(self, attrs: Any) -> Any:
                 )
         return super().validate(attrs)
 
+    def encrypt_sensitive_fields(self, validated_data: dict[str, Any]) -> dict[str, Any]:
+        # If any MODEL_ENCRYPTED_FIELDS are present, encrypt them
+
+        for field in self.Meta.model.MODEL_ENCRYPTED_FIELDS:
+            if field in validated_data:
+                validated_data[field] = encrypt(validated_data[field])
+        return validated_data
+
     def create(self, validated_data: dict[str, Any], **kwargs) -> Team:
         serializers.raise_errors_on_nested_writes("create", self, validated_data)
         request = self.context["request"]
         organization = self.context["view"].organization  # Use the org we used to validate permissions
 
+        validated_data = self.encrypt_sensitive_fields(validated_data)
```

Comment:
hmm, instead of doing it like this, which is a bit wonky / easy to forget to encrypt and then bork, better to take the personal API token approach, where we generate the shared secret, vs. depending on the user for it, and they can input it in their systems. And then we can reset it. This also implies we don't necessarily need encryption on these fields, since they're not coming from the user anyway.

Reply:
.. only reason I can think of not to use the personal API key as the secret is because it's at the org level, and per user, rather than per team.

Reply:
Curious what you think Ben, currently I'm thinking:

1. Keep the encryption
2. Move it to a separate patch request which is the only way to generate / reset this key, and encryption happens here

Reply:
For the demo will continue like it is right now

Reply:
I think we're being overly careful about it. Most UIs that I've seen with this kind of thing display it in plain text. 

As it's only a shared secret for signing the level of risk is low. Versus an api key that is actually an _authentication_ key which is what makes it super sensitive

Reply:
That said I think we should encrypt it because why not 😅 but I wouldn't stress about never making it displayable again